hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/diskbalancer/DiskBalancerService.java

```diff
@@ -367,7 +393,88 @@ private class DiskBalancerTask implements BackgroundTask {
 
     @Override
     public BackgroundTaskResult call() {
-      // TODO: Details of handling tasks
+      long containerId = containerData.getContainerID();
+      boolean destVolumeIncreased = false;
+      Path diskBalancerTmpDir = null, diskBalancerDestDir = null;
+      long containerSize = containerData.getBytesUsed();
+      try {
+        diskBalancerTmpDir = destVolume.getTmpPath()
+            .resolve(DISK_BALANCER_DIR).resolve(String.valueOf(containerId));
+
+        // Copy container to new Volume's tmp Dir
+        ozoneContainer.getController().copyContainer(containerData,
```

Comment:
I think currently we don't have such locks. Since the locks could be time-consuming for export and import, other operations might be complaining about it.

And I think it's possible now that there are already some replicas not same with others, due to DN restart or exporting, it should be good to have a mechanism to sync among all replicas.

Reply:
We have a design in progress that will deal with mis-matches between the containers in terms of deleted blocks, but I feel we should try to avoid causing such problems if we can. I am not sure how hard that might be with the current code layout.

Eg if there was a read lock held for the duration of the container move, then it would allow block reads etc to come through, but would stop block deletes. However at the moment, I don't think readChunk has any locks involved, so this would be a lot of difficult changes I think.

Reply:
> We have a design in progress that will deal with mis-matches between the containers in terms of deleted blocks

Looking forward to this feature.

Currently I think the read won't be affected, clients can still read data while the move is ongoing.

Reply:
I think it will be ok if the block deletes get missed. This isn't really any difference to replication between nodes, as after tarring a container a block could be deleted before the new replica is available on the target node.